### PR TITLE
Re-Add probably stripped Methods to NBT-Classes

### PR DESCRIPTION
**Pull request by TechnicianLP**
Specificall readds:
`getLongArray()` to `NBTTagLongArray` (to actually make it usable without reflection)
missing `long[]` getters and setters to `NBTTagComound`
missing getters to `NBTTagList` (didnt add `set()` or `append()` here as vanilla doesnt have any overloads here by default - tell me if those should be added as well)

### CODE REVIEW

**Comment by TechnicianLP** [comment]
Yes or No?

File: patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch [hunk 1]
```diff
@@ -16,3 +16,29 @@
...
  
          try
+@@ -533,4 +535,25 @@
+     {
+         return field_193583_c.matcher(p_193582_0_).matches() ? p_193582_0_ : NBTTagString.func_193588_a(p_193582_0_);
+     }
++
++    // FORGE START
++
++    public void setLongArray(String key, long[] value)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch]: Should be `putLongArray`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 2]
```diff
@@ -9,9 +9,92 @@
...
++
++    // Forge START
++
++    @Override
++    public java.util.Iterator<NBTBase> iterator()
++    {
++        return field_74747_a.iterator();
++    }
++
++    public byte getByteAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getByte`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 3]
```diff
@@ -9,9 +9,92 @@
...
++        }
++        return 0;
++    }
++
++    public boolean getBooleanAt(int i)
++    {
++        return getByteAt(i) != 0;
++    }
++
++    public short getShortAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getShort`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 4]
```diff
@@ -9,9 +9,92 @@
...
++        if (i >= 0 && i < this.field_74747_a.size())
++        {
++            NBTBase nbtbase = this.field_74747_a.get(i);
++            if (nbtbase.func_74732_a() == 1)
++                return ((NBTTagByte)nbtbase).func_150290_f();
++        }
++        return 0;
++    }
++
++    public boolean getBooleanAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getBoolean`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 5]
```diff
@@ -9,9 +9,92 @@
...
++        if (i >= 0 && i < this.field_74747_a.size())
++        {
++            NBTBase nbtbase = this.field_74747_a.get(i);
++            if (nbtbase.func_74732_a() == 2)
++                return ((NBTTagShort)nbtbase).func_150289_e();
++        }
++        return 0;
++    }
++
++    public long getLongAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getLong`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 6]
```diff
@@ -9,9 +9,92 @@
...
++        if (i >= 0 && i < this.field_74747_a.size())
++        {
++            NBTBase nbtbase = this.field_74747_a.get(i);
++            if (nbtbase.func_74732_a() == 4)
++                return ((NBTTagLong)nbtbase).func_150291_c();
++        }
++        return 0;
++    }
++
++    public byte[] getByteArrayAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getByteArray`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 7]
```diff
@@ -9,9 +9,92 @@
...
++                NBTTagList nbttaglist = (NBTTagList) nbtbase;
++                if (!nbttaglist.func_82582_d() && nbttaglist.func_150303_d() != type)
++                    return new NBTTagList();
++                return nbttaglist;
++            }
++        }
++        return new NBTTagList();
++    }
++
++    public long[] getLongArrayAt(int i)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getLongArray`.

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 8]
```diff
@@ -9,9 +9,92 @@
...
++        if (i >= 0 && i < this.field_74747_a.size())
++        {
++            NBTBase nbtbase = this.field_74747_a.get(i);
++            if (nbtbase.func_74732_a() == 7)
++                return ((NBTTagByteArray)nbtbase).func_150292_c();
++        }
++        return new byte[0];
++    }
++
++    public NBTTagList getListTagAt(int i, int type)
```
Comment (kashike) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: Should be `getList`.

**Review by kashike — CHANGES_REQUESTED** [review]

**Comment by MajorTuvok** [comment]
Waiting for changes...
Not stale...

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch (hunk 4 above)
Comment (kashike) [reply patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: ^

File: patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch [hunk 9]
```diff
@@ -9,9 +9,92 @@
...
++    {
++        return field_74747_a.iterator();
++    }
++
++    public byte getByte(int i)
++    {
++        if (i >= 0 && i < this.field_74747_a.size())
++        {
++            NBTBase nbtbase = this.field_74747_a.get(i);
++            if (nbtbase.func_74732_a() == 1)
```
Comment (tterrag1098) [inline patches/minecraft/net/minecraft/nbt/NBTTagList.java.patch]: No need to hardcode these constants in our own code, use `Constants.NBT` instead

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

**Comment by TechnicianLP** [comment]
I will be able to take a look at this again in a few days

File: patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch [hunk 10]
```diff
@@ -24,3 +24,29 @@
...
++    public void putLongArray(String key, long[] value)
++    {
++        this.field_74784_a.put(key, new NBTTagLongArray(value));
++    }
++
++    public long[] getLongArray(String key)
++    {
++        try
++        {
++            if (this.func_150297_b(key, 12))
```
Comment (mezz) [inline patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch]: use `net.minecraftforge.common.util.Constants.NBT#TAG_LONG_ARRAY`

File: patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch [hunk 11]
```diff
@@ -24,3 +24,29 @@
...
++                return ((NBTTagLongArray) this.field_74784_a.get(key)).getLongArray();
++        }
++        catch (ClassCastException classcastexception)
++        {
++            throw new ReportedException(this.func_82581_a(key, 12, classcastexception));
```
Comment (mezz) [inline patches/minecraft/net/minecraft/nbt/NBTTagCompound.java.patch]: use `net.minecraftforge.common.util.Constants.NBT#TAG_LONG_ARRAY`

**Review by mezz — CHANGES_REQUESTED** [review]

**Review by mezz — APPROVED** [review]

**Pull request closed without merging**